modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/Redeployer.java

```diff
@@ -59,7 +61,10 @@ boolean deploymentExists() {
 
     Result redeployOnServer() {
         Operation op = new Operation("full-replace-deployment", new Address());
-        op.addAdditionalProperty("name", runtimeName);
+        op.addAdditionalProperty("name", name);
+        if (runtimeName != null) {
+            op.addAdditionalProperty("runtime-name", runtimeName);
```

Comment:
@josejulio I added the runtime-parameter here, if it is specified (not-null) I'll pass as a parameter of the full-replace-deployment op.